Remove /core docs

WillMoggridge opened the pull request:
Remove references to /core docs as they are now served inside their own
app, which is bundled in the repository with the docs.
This is live and serving the docs on docs.ubuntu.com/core

## QA

``` bash
./run clean
./run
```

Make sure it builds and the site loads correctly. Including checking <127.0.0.1:8007/maas>

Visit <127.0.0.1:8007/core>, see that it 404s

tbille commented:
* you forgot to remove the core in the `clean` script in the `package.json`:
`"clean": "rm -rf node_modules yarn-error.log css static/css *.log *.sqlite _site/ build/ .jekyll-metadata .bundle docs/ templates/landscape/ templates/documentation-builder/ templates/conjure-up/ templates/core/ templates/maas/ static/media"`

* remove core here to: https://github.com/canonical-websites/docs.ubuntu.com/blob/master/build-docs.sh#L48

I took the example from what @nottrobin did for the phone docs: https://github.com/canonical-websites/docs.ubuntu.com/pull/120

nottrobin commented:
Yes, the really important thing to do is to remove core from the `build-docs.sh` script

WillMoggridge commented:
I missed package.json but I could have the first thing I did was remove the build job... Oh well. I updated them now. Thanks for the spot.

I havn now also removed references to git-repo that I found.

nottrobin commented on README.md:
```diff
@@ -10,10 +10,10 @@ A central platform for hosting Ubuntu's various documentation portals.
 
 First [install Docker](https://www.docker.com/community-edition#/download). Linux users may need to add their user to the `docker` group.
 
-Then install [git-repo](https://github.com/esrlabs/git-repo) and [documentation-builder](https://github.com/canonicalltd/documentation-builder). E.g.:
```
This file looks weird. I think the ```````` code blocks need to be moved

nottrobin replied:
See: https://github.com/WillMoggridge/docs.ubuntu.com/blob/691e1f7ec439365bcc78c7609327c237d02326f1/README.md

WillMoggridge replied:
Pushed a change now, think it fixes it

nottrobin commented on README.md:
````diff
@@ -10,10 +10,10 @@ A central platform for hosting Ubuntu's various documentation portals.
-`` bash
-sudo snap install git-repo documentation-builder
+``` bash
+sudo snap install documentation-builder
````
Actually I think this line is erroneous. `build-docs.sh` is run through `package.json`, so it's running inside the `canonicalwebteam/dev` image. And it gets there because `documentation-builder` is in `requirements.txt`.

So we should remove this step from the readme.
